### ISSUE TITLE
Data table should have a <caption> element

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#1578]
## Environment
Chrome 70
macOS Mojave

## Problem/Violation
Screen reader users would benefit from a `<caption>` element on the data table describing the tables contents.

## Reference
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/caption

**Comment by joshblack** [comment]
What kind of conventions are typical for this?

**Comment by dakahn** [comment]
A summary of the information in the table. There's a write up in the reference link, but something like this:

![screen shot 2019-01-07 at 12 01 23 pm](https://user-images.githubusercontent.com/40970507/50784691-07549200-1274-11e9-8c9e-aa0fb9c727f0.png)

**Comment by joshblack** [comment]
@dakahn wouldn't this be up to teams to include? `DataTable` is a state container so it just accepts a render prop.

**Comment by dakahn** [comment]
Closing!
